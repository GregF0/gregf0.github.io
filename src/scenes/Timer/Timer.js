import React from 'react'
import Box from '@material-ui/core/Box'
import { CountdownCircle } from './components/CountdownCircle'
import { ToggleButton } from './components/ToggleButton'
import { RoundsCounter } from './components/RoundsCounter'
import { ResetButton } from './components/ResetButton'
import { SkipButton } from './components/SkipButton'
import { FullscreenDialog } from './components/Labels/FullscreenDialog'
import { DesktopDialog } from './components/Labels/DesktopDialog'
import { LabelButton } from './components/Labels/LabelButton'
import { useMediaQuery, useTheme } from '@material-ui/core'

export const Timer = () => {
  const theme = useTheme()

  const isMD = useMediaQuery(theme.breakpoints.up('md'))
  const isLG = useMediaQuery(theme.breakpoints.up('lg'))
  const isXL = useMediaQuery(theme.breakpoints.up('xl'))

  const getCircleSize = () => {
    if (isXL) return 350
    if (isLG) return 300
    if (isMD) return 250

    return 200
  }

  return (
    <Box width={getCircleSize()} m="auto">
      <LabelButton />
      <CountdownCircle />

      <Box display="flex" justifyContent="center" alignItems="center" my={3}>
        <ResetButton />

        <Box display="flex" flexDirection="column" alignItems="center" mx={3}>
          <ToggleButton />
        </Box>

        <SkipButton />
      </Box>

      <Box display="flex" justifyContent="center">
        <RoundsCounter />
      </Box>

      <FullscreenDialog />
      <DesktopDialog />
    </Box>
  )
}
