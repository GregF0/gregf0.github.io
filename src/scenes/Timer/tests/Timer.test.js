import React from 'react'
import { createShallow } from '@material-ui/core/test-utils'
import { Timer } from '../Timer'

describe('<SkipButton />', () => {
  const shallow = createShallow()
  const wrapper = shallow(<Timer />)

  test('should render <Timer /> correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
