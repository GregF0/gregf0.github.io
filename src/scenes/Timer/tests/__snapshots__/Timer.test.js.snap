// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkipButton /> should render <Timer /> correctly 1`] = `
<Styled(MuiBox)
  m="auto"
  width={200}
>
  <LabelButton />
  <CountdownCircle />
  <Styled(MuiBox)
    alignItems="center"
    display="flex"
    justifyContent="center"
    my={3}
  >
    <ResetButton />
    <Styled(MuiBox)
      alignItems="center"
      display="flex"
      flexDirection="column"
      mx={3}
    >
      <ToggleButton />
    </Styled(MuiBox)>
    <SkipButton />
  </Styled(MuiBox)>
  <Styled(MuiBox)
    display="flex"
    justifyContent="center"
  >
    <RoundsCounter />
  </Styled(MuiBox)>
  <FullscreenDialog />
  <DesktopDialog />
</Styled(MuiBox)>
`;
